lib/broadway/producer.ex

```diff
@@ -236,4 +277,27 @@ defmodule Broadway.Producer do
             "struct. In case you're using a standard GenStage producer, please set the " <>
             ":transformer option to transform produced events into message structs"
   end
+
+  defp maybe_rate_limit_and_buffer_messages(state, messages) do
+    if state[:rate_limiting_enabled?] && messages != [] do
```

Comment:
It is ok to always have this field in the state. Then we can go ahead and pattern match on it.

Reply:
This was actually the tests' fault 😄